robot/packages/cloud_package_manager_test.go

```diff
@@ -200,14 +200,33 @@ func TestCloud(t *testing.T) {
 
 		err = pm.Sync(ctx, input)
 		test.That(t, err, test.ShouldNotBeNil)
-		test.That(t, err.Error(), test.ShouldContainSubstring, "download did not match expected hash")
+		test.That(t, err.Error(), test.ShouldContainSubstring, "failed to decode")
 
 		err = pm.Cleanup(ctx)
 		test.That(t, err, test.ShouldBeNil)
 
 		validatePackageDir(t, packageDir, []config.PackageConfig{})
 	})
 
+	t.Run("leading zeroes checksum", func(t *testing.T) {
+		packageDir, pm := newPackageManager(t, client, fakeServer, logger)
+		defer utils.UncheckedErrorFunc(func() error { return pm.Close(context.Background()) })
+
+		fakeServer.SetChecksumWithLeadingZeroes(true)
```

Comment:
fakeServer is reused by other tests -- to avoid impacting them, can you clear this in a Cleanup handler?

T.Cleanup [docs here](https://pkg.go.dev/testing#T.Cleanup)

Reply:
Is this handled in the test file in `Clear`?

Reply:
ah you're right, it's creating a new one for each test